README.md

```diff
@@ -0,0 +1,222 @@
+are specified with a `name`, a `value`, and an optional `state` or `previous`
+which controls the behavior.  For example:
+```yaml
+kernel_settings_parameters:
+  sysctl:
+    - name: fs.epoll.max_user_watches
+      value: 785592
+    - name: fs.file-max
+      value: 379724
+  sysfs:
+    - name: /sys/kernel/debug/x86/pti_enabled
+      value: 0
+    - name: /sys/kernel/debug/x86/retp_enabled
+      value: 0
+    - name: /sys/kernel/debug/x86/ibrs_enabled
+      value: 0
+  vm:
+    - name: transparent_hugepages
+      value: madvise
+  bootloader:
+    - name: cmdline
+      value:
+        - name: spectre_v2
+          value: "off"
+        - name: nopti
+```
+See the `tuned` documentation above for the complete list of plugins `sysctl`,
+`sysfs`, etc., and the format of settings for those plugins.  *NOTE* that
+these settings are **additive**.  That is, they are applied **in addition to**
+any current settings.  For example, if you already had
+```yaml
+  sysctl:
+    - name: kernel.threads-max
+      value: 29968
+    - name: vm.max_map_count
+      value: 65530
+```
+then after applying the above, you would have
+```yaml
+  sysctl:
+    - name: kernel.threads-max
+      value: 29968
+    - name: vm.max_map_count
+      value: 65530
+    - name: fs.epoll.max_user_watches
+      value: 785592
+    - name: fs.file-max
+      value: 379724
+  ...
+```
+This allows multiple higher level roles or playbooks to use this role to
+provide the kernel settings specific to that component.  For example, if you
+are installing a web server and a database server on the same machine, and
+they both require setting kernel parameters, the `kernel_settings` role allows
+you to set them both.
+
+If you specify multiple settings with the same name in a section, the last
+one will be used.
+
+If you want to replace *all* of the settings in a section with your supplied
+values, use `previous: replaced` as a single, preferably first element in the
+list of settings.  This indicates that the `previous` settings in the system
+should be `replaced` with the given settings.  For example:
+```yaml
+  sysctl:
+    - previous: replaced
+    - name: kernel.threads-max
+      value: 30000
+    - name: vm.max_map_count
+      value: 50000
+```
+This will have the effect of removing all of the existing settings for
+`sysctl`, and adding the specified settings.
+If you want to remove a single setting, specify `state: absent` in the
+individual setting, instead of a `value`:
+```yaml
+  sysctl:
+    - name: kernel.threads-max
+      value: 30000
+    - name: vm.max_map_count
+      state: absent
+```
+This will remove the `vm.max_map_count` setting from the `sysctl` section.
+If you want to remove an entire section, specify `state: empty` as a `dict`
+instead of a `list`:
+```yaml
+  sysctl:
+    state: empty
+```
+This will have the effect of removing all of the `sysctl` settings.
+
+The `bootloader` `cmdline` value is a special case.  There may be situations
```

Comment:
Why is it special? To me it looks similar to the above.

Reply:
(except that it is nested one level below bootloader, while in case of the above they are on the toplevel. But otherwise `cmdline` seems to behave in a pretty similar way as `sysctl` does.)

Reply:
The `bootloader` `cmdline` value is a list.  Every other value at the same level is a scalar.